### PR TITLE
feat: update contributing.md document

### PR DESCRIPTION
**Pull request by AlexTheWizardL**
This PR covers the update of contributing file based on the new code style and best practices applied to the code.

@lasuk Please review the changes and provide feedback or approval for merging.

### CODE REVIEW

**Comment by lasuk** [comment]
Thanks for those changes. This would be a good moment to incorporate #25 and the [first comment](https://github.com/macxred/cashctrl_ledger/issues/53#issuecomment-2191171479) in macxred/cashctrl_ledger#53. Your thoughts, @AlexTheWizardL?

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -85,28 +84,27 @@ Follow these steps to set up your local development environment on Unix/Mac OS:
...
 
-### Commit description
+### Commit Description
 ```
 tickets: #taskId, #taskId
 Ex.: tickets: #1, #2, #3
 ```
 
-### Pull-request name
+### Pull Request Name
```
Comment (lasuk) [inline CONTRIBUTING.md:107]: This is a good moment to review the naming patterns introduced in b5c5ee8:
- With your present experience, do all pattern still make sense?
- Do the patterns align with our role models, the big open source projects?

**Review by lasuk — COMMENTED** [review]
Thanks for revising the contribution guidelines.

The current stage is a step forward, but needs a bit more care towards clarification and integration of differing sections.

File: CONTRIBUTING.md (hunk 1 above)
Comment (AlexTheWizardL) [reply CONTRIBUTING.md:107]: ## Reviewing Existing Naming Patterns

### Branch Name
After reviewing common naming patterns and consulting ChatGPT, I've concluded the following:

Current Pattern: 
```
(feat|fix|docs|style|refactor|test|revert)/taskId_task-short-description
```
This is aligned with common practices. I usually use this pattern, but there's a preference to create branches by dropping the `(feat|fix|docs|style|refactor|test|revert)/taskId_` part.

**Options:**
- [ ] Leave as it is
- [ ] Drop prefix and taskID

### Commit Message
This section should combine the **Commit Name** and **Commit Description** sections, aligning with [issue#25](https://github.com/macxred/cashctrl_api/issues/25).

**Proposed Format:**
```
<Summarize change(s) in around 50 characters or less>

<More detailed explanatory description of the change wrapped into about 72 characters>
```
**Example:**
```
feat: add user authentication

Implement OAuth2 authentication to enhance security and user convenience. Closes #123.
```

### Pull Request Name
Current Pattern:
```
(Feat|Fix|Docs|Style|Refactor|Test|Revert): #taskId description
```
This aligns with common practices. Sometimes we use this pattern, but often we just mention a short description that the PR covers.

**Options:**
- [ ] Leave as it is
- [ ] Use the following pattern:
```
<Summarize Pull Request purpose in around 50 characters or less>
```
**Example:**
```
Add user authentication
```

**Comment by lasuk** [comment]
Asking ChatGPT for brand naming conventions of open source Python projects hosted on GitHub reveals that most projects use a prefix, but not a task id.

<details> <summary> Click for conversation with ChatGPT </summary>

### Discussion on Git Branch Naming Conventions for Python Open Source Projects

#### Question:
What naming conventions do Python open source projects hosted on GitHub use for git branches? Please add examples of a few large Python open source projects.

#### Answer:
Naming conventions for git branches in Python open source projects on GitHub vary, but some common patterns emerge. These conventions help maintain organization and clarity within the project. Here are some typical conventions along with examples from large Python open source projects:

##### Common Naming Conventions

1. **Main Branches:**
   - `main` or `master`: The primary branch where the source code of HEAD always reflects a production-ready state.
   - Example: `main`

2. **Feature Branches:**
   - `feature/<description>`: Used for developing new features.
   - Example: `feature/authentication-module`

3. **Bugfix Branches:**
   - `bugfix/<description>`: Used for fixing bugs.
   - Example: `bugfix/fix-login-issue`

4. **Release Branches:**
   - `release/<version>`: Used to prepare for a new release.
   - Example: `release/2.0.0`

5. **Hotfix Branches:**
   - `hotfix/<description>`: Used for urgent fixes in production.
   - Example: `hotfix/security-patch`

6. **Experimental Branches:**
   - `experiment/<description>`: Used for experimental features or ideas.
   - Example: `experiment/new-algorithm`

##### Examples from Large Python Open Source Projects

1. **Django**
   - Main Branch: `main`
   - Example feature branch: `feature/async-support`
   - Example bugfix branch: `bugfix/issue-12345`
   - Example release branch: `release/3.2`

2. **Flask**
   - Main Branch: `main`
   - Example feature branch: `feature/rest-api`
   - Example bugfix branch: `bugfix/template-rendering`
   - Example release branch: `release/2.1.0`

3. **Pandas**
   - Main Branch: `main`
   - Example feature branch: `feature/dataframe-enhancements`
   - Example bugfix branch: `bugfix/groupby-errors`
   - Example release branch: `release/1.3.0`

4. **Scikit-Learn**
   - Main Branch: `main`
   - Example feature branch: `feature/neural-networks`
   - Example bugfix branch: `bugfix/cross-validation-issue`
   - Example release branch: `release/0.24.2`

5. **NumPy**
   - Main Branch: `main`
   - Example feature branch: `feature/matrix-multiplication`
   - Example bugfix branch: `bugfix/array-indexing`
   - Example release branch: `release/1.21.0`

##### Conclusion
These conventions help keep branches organized and ensure that the purpose of each branch is clear. While the specific names might vary slightly between projects, the general patterns of naming branches based on their purpose (feature, bugfix, release, hotfix, etc.) are widely followed.

</details>

**Comment by lasuk** [comment]
Merging. Thanks for updating the guidelines!

**Pull request merged**